telemetry/serializer_test.go

```diff
@@ -208,6 +214,7 @@ var _ = Describe("BinarySerializer", func() {
 		result, _ = bs.Deserialize(msgBytes, "Socket-42")
 		bs.Dispatch(result)
 		Expect(CallbackTester.counter).To(Equal(1))
+		Expect(CallbackTester.errors).To(Equal(0))
```

Comment:
Can we test when ReportError gets called?

Reply:
Not on this test - but can we test the case that does

Reply:
would be hard to simulate this scenario in tests. Will prefer keeping the testing as is.